src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts

```diff
@@ -205,13 +213,38 @@ export class MarkdownToolbarComponent extends AngularDisposable {
 		}
 	}
 
-	public onInsertButtonClick(event: MouseEvent, type: MarkdownButtonType): void {
-		let go = new MarkdownTextTransformer(this._notebookService, this.cellModel, this._instantiationService);
-		let trigger = event.target as HTMLElement;
-		go.transformText(type, trigger);
+	public async onInsertButtonClick(event: MouseEvent, type: MarkdownButtonType): Promise<void> {
+		event.preventDefault();
+		event.stopPropagation();
+		let triggerElement = event.target as HTMLElement;
+		let needsTransform = true;
+		let calloutResult: ILinkCalloutDialogOptions;
+		if (type === MarkdownButtonType.LINK_PREVIEW) {
+			calloutResult = await this.createCallout(type, triggerElement);
+			// If no URL is present, no-op
+			if (!calloutResult.insertLinkUrl) {
+				return;
+			}
+			// If cell edit mode isn't WYSIWYG, use result from callout. No need for further transformation.
+			if (this.cellModel.currentMode !== CellEditModes.WYSIWYG) {
+				needsTransform = false;
+			} else {
+				// Otherwise, re-focus on the output element, and insert the link directly.
+				this.output?.nativeElement?.focus();
+				// Callout is responsible for returning escaped strings
+				document.execCommand('insertHTML', false, `<a href="${calloutResult?.insertLinkUrl}">${calloutResult?.insertLinkLabel}</a>`);
```

Comment:
Are these commands documented anywhere? And what makes it so that the text is going to be inserted in the correct position - does the output element remember the cursor position it was at?

Reply:
https://www.w3schools.com/jsref/met_document_execcommand.asp

Reply:
Why did we choose to use these functions instead of functions on the VS wrappers over this stuff? Was there some functionality that you couldn't get through VS Editors alone or something?

For example for this - we add text to editors all the time. Why are we calling this special command to do that for us here?

Reply:
Output components aren't editors, and aren't webviews. They're just div's.

Reply:
We can talk about whether that's the right long term design but notebook outputs were previously webviews, which caused some pretty extreme bad perf. In vscode, they have a solution where only webview is created for a notebook, and it's shared between outputs (but has some builtin isolation)